marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs

```diff
@@ -503,3 +521,22 @@ failWhenFull x = do
   if isEmpty
   then pure ()
   else error "an indexer raised an error"
+
+drainUntil :: C.ChainPoint -> TChan (ChainSyncEvent (BlockInMode CardanoMode)) -> QSemN -> IO ()
```

Comment:
Add a comment here on what that function does and what it is used for.

Even better, add a {- [NOTE] ...} somewhere explaining the choice we made on handling resuming and link to it here.

Reply:
b79ce37bf378295f1d651e9dd12876a74dbe93a8